reconcile/cli.py

```diff
@@ -402,6 +428,58 @@ def run_integration(func_container, ctx, *args, **kwargs):
                 f.write(json.dumps(gqlapi.get_queried_schemas()))
 
 
+def early_exit_integration(
+    int_name: str, compare_sha: str, func_container, *args, **kwargs
+) -> bool:
+    early_exit_desired_state_function = "early_exit_desired_state"
+    # does the integration support early exit?
+    if "early_exit_desired_state" not in dir(func_container):
+        logging.warning(
+            f"{int_name} does not support early exit. it does not offer a "
+            f"function called {early_exit_desired_state_function}"
+        )
+        return False
+
+    # get desired state from comparison bundle
+    try:
+        gql.init_from_config(
+            sha=compare_sha,
+            integration=int_name,
+            validate_schemas=True,
+            print_url=True,
+        )
+        previous_desired_state = func_container.early_exit_desired_state(
+            *args, **kwargs
+        )
+    except Exception:
+        logging.exception(
+            f"Failed to fetch desired state for comparison bundle {compare_sha} failed"
+        )
+        return False
+
+    # get desired state from current bundle
+    try:
+        gql.init_from_config(
+            autodetect_sha=True,
+            integration=int_name,
+            validate_schemas=True,
+            print_url=True,
+        )
+        current_desired_state = func_container.early_exit_desired_state(*args, **kwargs)
+    except Exception:
+        logging.exception("Failed to fetch desired state for current bundle failed")
+        return False
+
+    # compare
+    from deepdiff import DeepDiff
+
+    diff = DeepDiff(previous_desired_state, current_desired_state, ignore_order=True)
+    if diff:
```

Comment:
nit: `return not diff` ?

Reply:
[29dea50](https://github.com/app-sre/qontract-reconcile/pull/2672/commits/29dea5026825be99a02c48411d1cce88f62c932b)